### ISSUE TITLE
Options Box

### ISSUE THREAD

**Issue by 2bigO** [assister-ai/assister#8]
Parent issue: #7

**Comment by mehdibalouchi** [comment]
hi @keyvan-m-sadeghi 
would you please give an explanation about what happens after clicking `Undo` and `Redo`?

**Comment by 2bigO** [comment]
@mehdibalouchi `OptionsBox` is a dynamic box, its options change based on the current context. When a message is selected for example, it offers the option of `run again`, much later on we might explore things like `reply` when we figure out whether such semantics exist (based on user input).

When nothing is selected, `undo` and `redo` seem like what the user would expect. I'm still not sure about the `TFx` side of `undo` and `redo` design. I'm very much in favour of not adding any magic; undo and redo would be normal commands that are associated with these buttons if present. Implementing general undo and redo is a well known science, see [the redux version](https://redux.js.org/recipes/implementing-undo-history) for example.

On the other hand, more complex application logic, especially those involving side effects, might require more fine grained control in a per command setup. Say, a command updates a record in a database, general `undo` would not address this use case.

To summarise, it's a WIP on the TFx side, though the GUI can assume it is a given.
